### PR TITLE
docs(examples): add an example on how to use `deno_kv_oauth`

### LINKED ISSUE #1218: proposal(docs): add an example of how to use `deno_kv_oauth` module
The module can be found here: https://deno.land/x/deno_kv_oauth
Perhaps, it'd be best to not do this example until the API is stabilised.

### PR DESCRIPTION
**Pull request by mbhrznr**
this pr adds an example on how to use `deno_kv_oauth` to fresh's docs.  
closes #1218

### CODE REVIEW

**Comment by mbhrznr** [comment]
@iuioiua as promised, here are pr and ping :)

**Comment by lino-levan** [comment]
Is the Deno KV OAuth API stable?

**Review by iuioiua — CHANGES_REQUESTED** [review]
Please run `deno fmt`

**Review by iuioiua — CHANGES_REQUESTED** [review]
The PR in its current state was taken from Deno KV OAuth's README, which uses a generic HTTP server. Could you please use Fresh's method of creating custom handlers? Documentation [here](https://fresh.deno.dev/docs/getting-started/custom-handlers).

**Comment by iuioiua** [comment]
> Is the Deno KV OAuth API stable?

Not yet. But it soon will be. We'll note its beta status until then.

File: docs/examples/using-deno-kv-oauth.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+This example uses GitHub as the OAuth 2.0 provider. However there is a suite of
+[pre-configured providers](https://deno.land/x/deno_kv_oauth#pre-configured-oauth-20-clients)
+available.
+
+1. Register a [new GitHub OAuth](https://github.com/settings/applications/new)
+   application, if you haven't already.
+2. Create your pre-configured OAuth client instance. For reusability the
+   instance is stored in `utils/oauth-client.ts`.
+
+```typescript
```
Comment (iuioiua) [inline docs/examples/using-deno-kv-oauth.md]:
```suggestion
```ts
```
Could you please do this for all other TypeScript snippets? Just a matter of consistency is all.

File: docs/examples/using-deno-kv-oauth.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import type { Handlers } from "$fresh/server.ts";
+import { signIn } from "https://deno.land/x/deno_kv_oauth/mod.ts";
+import { oauth2Client } from "../../utils/oauth-client.ts";
+
+export const handler: Handlers = {
+  async GET(req) {
+    return await signIn(req, oauth2Client);
+  },
+};
+```
```
Comment (iuioiua) [inline docs/examples/using-deno-kv-oauth.md]: Could you please indent snippets that are part of a numbered item to help the look and feel?

File: docs/examples/using-deno-kv-oauth.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+```typescript
+// routes/auth/signout.ts
+import type { Handlers } from "$fresh/server.ts";
+import { signOut } from "https://deno.land/x/deno_kv_oauth/mod.ts";
+
+export const handler: Handlers = {
+  async GET(req) {
+    return await signOut(req, "/");
```
Comment (iuioiua) [inline docs/examples/using-deno-kv-oauth.md]:
```suggestion
    return await signOut(req);
```
This parameter is `/` by default anyway.

File: docs/examples/using-deno-kv-oauth.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+export const handler: Handlers = {
+  async GET(req) {
+    // `accessToken` and `sessionId` can be used for further consumption
+    const { accessToken, response, sessionId } = await handleCallback(
+      req,
+      oauth2Client,
+      "/", // an optional `redirectUrl` can be passed
+    );
+
```
Comment (iuioiua) [inline docs/examples/using-deno-kv-oauth.md]:
```suggestion
    // Return object also includes `accessToken` and `sessionId` properties.
    const { response } = await handleCallback(req, oauth2Client);
    return response;
```

File: docs/examples/using-deno-kv-oauth.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+[pre-configured providers](https://deno.land/x/deno_kv_oauth#pre-configured-oauth-20-clients)
+available.
+
+1. Register a [new GitHub OAuth](https://github.com/settings/applications/new)
+   application, if you haven't already.
+2. Create your pre-configured OAuth client instance. For reusability the
+   instance is stored in `utils/oauth-client.ts`.
+
+```typescript
+// utils/oauth-client.ts
```
Comment (iuioiua) [inline docs/examples/using-deno-kv-oauth.md]:
```suggestion
// utils/oauth2_client.ts
```

File: docs/examples/using-deno-kv-oauth.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+```
+
+3. Using the OAuth 2.0 client instance, insert the authentication flow functions
+   into your authentication routes. In this example, there are dedicated handler
+   routes at `routes/auth/signin.ts`, `routes/auth/signout.ts` and
+   `routes/auth/callback.ts`. Please ensure that the `callback` handler matches
+   the Authorization callback URL in the configured OAuth application!
+
+```typescript
+// routes/auth/signin.ts
```
Comment (iuioiua) [inline docs/examples/using-deno-kv-oauth.md]:
```suggestion
// routes/signin.ts
```
We want the user to do the bare minimum to get OAuth working. Flattening the route structure means one less folder that the user has to create.

File: docs/examples/using-deno-kv-oauth.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+      },
+    });
+    const user = await response.json();
+
+    console.log(user);
+  }
+
+  return await ctx.next();
+}
+```
```
Comment (iuioiua) [inline docs/examples/using-deno-kv-oauth.md]: Can we instead put this into the home route `/`?

**Review by iuioiua — CHANGES_REQUESTED** [review]
Thanks for this! We should aim to make this as simple and lean as possible. The question should be: what are the minimum requirements to get this module working with Fresh? We can add a few short bullet points at the end of the article on how else the module can be used. E.g. environment variables, custom OAuth client, and other providers. Then, we can point the reader to the module to discover how they can extend said functionality.

File: docs/examples/using-deno-kv-oauth.md (hunk 2 above)
Comment (mbhrznr) [reply docs/examples/using-deno-kv-oauth.md]: initially added some indentation, yet when running `deno fmt` it gets removed again for the code snippets.

File: docs/examples/using-deno-kv-oauth.md (hunk 7 above)
Comment (mbhrznr) [reply docs/examples/using-deno-kv-oauth.md]: absolutely! added a more useful example here.

**Comment by mbhrznr** [comment]
@iuioiua thanks again for the thorough review! 🙏 
let me know if there's still something to improve!

**Review by iuioiua — APPROVED** [review]
I made a few tweaks, and now, LGTM! Thanks for this.

Tip: double-indent snippets. `deno fmt` will then re-align the snippets to the numbered text.

**Review by marvinhagemeister — APPROVED** [review]

**Review by iuioiua — APPROVED** [review]

**Pull request merged**